src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets

```diff
@@ -418,6 +418,24 @@ Copyright (c) .NET Foundation. All rights reserved.
     </ItemGroup>
   </Target>
 
+  <Target Name="ResolveOffByDefaultAnalyzers" AfterTargets="ResolveTargetingPackAssets"
+          Condition="'@(FrameworkReference)' != ''">
+    <ItemGroup>
+      <OffByDefaultAnalyzer Include="Microsoft.AspNetCore.Http.Generators"
+                            IsEnabled="$(EnableRequestDelegateGenerator)"/>
+    </ItemGroup>
+
+    <ItemGroup>
+      <AnalyzerToRemove Include="@(OffByDefaultAnalyzer)"
+                        Path="%(Analyzer.Identity)"
+                        Condition="'%(Analyzer.Filename)' == '@(OffByDefaultAnalyzer)'" />
```

Comment:
There are tests covering this, right?  The join is a bit tricky and I probably wouldn't get it right on the first try.

Reply:
FWIW here's the current logic for handling analyzers in NuGet packages in the runtime (to provide the generator to folks using older SDKs) - https://github.com/dotnet/runtime/blob/1e601c5101e1f0573cd03560615aa8f6cb38fa64/eng/packaging.targets#L196-L229

 Is there any concern around these targets conflicting with each other?

Reply:
> There are tests covering this, right? The join is a bit tricky and I probably wouldn't get it right on the first try.

Yep, there are pre-existing tests for the behavior in [this test class](https://github.com/dotnet/sdk/blob/28ba4ed390c790662da8a89309e311b1ce900f39/src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToUseAnalyzers.cs#L26-L60) which is how I validated it worked (definitely not on the first try though 🤪).

> Should this work on analyzers coming from NuGet packages too? 

At least for this PR, I was focused on the case for analyzers derived from targeting packs as described here (https://github.com/dotnet/sdk/issues/30872#issuecomment-1463030722).

> Is there any concern around these targets conflicting with each other?

It doesn't seem like they would conflict, especially because the config generator has not yet been added to the list of `OffByDefaultAnalyzer`s (assuming this PR is merged).

I'm not sure if it is worthwhile to move the targets that are currently in the runtime to the SDK.

Reply:
I see, looks good.